File: app/src/main/java/vng/zalo/tdtai/zalo/zalo/ZaloApplication.java
package vng.zalo.tdtai.zalo.zalo;

import androidx.multidex.MultiDexApplication;

import com.google.firebase.firestore.FirebaseFirestore;

import vng.zalo.tdtai.zalo.zalo.models.UserInfo;

public class ZaloApplication extends MultiDexApplication{
    public static UserInfo currentUser;

    @Override
    public void onCreate() {
        super.onCreate();
//        ApplicationComponent applicationComponent = DaggerApplicationComponent
//                .builder()
//                .applicationModule(new ApplicationModule(this))
//                .build();
//        ApplicationComponent applicationComponent = DaggerApplicationComponent.create();
//        applicationComponent.inject(this);
//        firestore = FirebaseFirestore.getInstance();
    }

    public static FirebaseFirestore getFirebaseInstance(){
        return FirebaseFirestore.getInstance();
    }
}

File: app/src/main/java/vng/zalo/tdtai/zalo/zalo/ZaloApplication.kt
package vng.zalo.tdtai.zalo.zalo

import androidx.multidex.MultiDexApplication
import com.google.firebase.firestore.FirebaseFirestore
import com.google.firebase.storage.FirebaseStorage
import vng.zalo.tdtai.zalo.zalo.models.UserInfo

class ZaloApplication : MultiDexApplication() {

//    override fun onCreate() {
//        super.onCreate()
//        val applicationComponent = DaggerApplicationComponent
//                .builder()
//                .applicationModule(ApplicationModule(this))
//                .build()
//                ApplicationComponent applicationComponent = DaggerApplicationComponent.create();
//        applicationComponent.inject(this)
//    }

    companion object {
        var currentUser: UserInfo? = null

        val firebaseFirestore: FirebaseFirestore
            get() = FirebaseFirestore.getInstance()
        val firebaseStorage: FirebaseStorage
            get() = FirebaseStorage.getInstance()
    }
}